plugins/reach/plugins/OpenCalais/batch/kReachVendorTaskOpenCalaisProcessorEngine.php

```diff
@@ -59,7 +63,20 @@ function handleTask(KalturaEntryVendorTask $vendorTask)
         KBatchBase::impersonate($vendorTask->partnerId);
 
         try {
-            $mappingProfileId = $this->getMappingMetadataProfileId();
+            // If entry has PreventAutomatedCategorization set to True --> no tags should be created for the entry.
+            $autoCategorizationMetadataProfileId = $this->getMetaDataProfileId(self::SHOW_DATA_EXTRACTION_WORKFLOW_SYSTEM_NAME);
+            $preventAutoCategorizationMetadata = $this->retrieveMetadataObjectsByMetadataProfileAndObjectId($autoCategorizationMetadataProfileId, KalturaMetadataObjectType::ENTRY ,$vendorTask->entryId);
```

Comment:
i would check that this profile actually returned before passing to retrieveMetadataObjectsByMetadataProfileAndObjectId, to avoid the list call for a null metadata profile afterwards

Reply:
@inbal-ben-david  list requests never return null objects. Unless you're using a invalid KS, you'll just get an empty list response.